### ISSUE TITLE
Add rule for trailing code/assertions in a pytest.raises block

### ISSUE THREAD

**Issue by goodspark** [astral-sh/ruff#18341]
### Summary

If a unit-under-test is successfully raising an error inside a pytest.raises block, then any trailing code after that call will not be executed and is effectively doing nothing. And given that a function not raising when it should is already an erroneous condition, any other code/assertions are likely not testing the code in the expected state.

Example:
```py
def asd():
  raise ValueError('blah')

def test_asd():
  with pytest.raises(ValueError, match='blah'):
    asd()
    # Code after this line will only be run if the unit-under-test doesn't raise.

    assert something == something_else
```

Given detecting the actual code that should be raising isn't quite obvious, I don't think there's an easy auto-fix for this. But the 'fix' should be:

```py
with pytest.raises(ValueError, match='blah'):
  ... # Code before thing that raises
  ... # Code that raises
  # Anything else moved out of the context manager

assert something == something_else
```

As far as detecting this, the pattern is probably something like:
> Any contiguous block of assertions _at the end_ of a pytest.raises block.

**Comment by MichaReiser** [comment]
I think this is already covered by https://docs.astral.sh/ruff/rules/pytest-raises-with-multiple-statements/  ([playground](https://play.ruff.rs/47ae2fe7-c11b-468c-96bb-ccf9608e6f84))

**Comment by goodspark** [comment]
Ah, you're right! Does exactly what I want. Thanks!
